File: srui-core/src/clipboard.rs
//! Clipboard abstraction — injected by the host per the sans-IO rule.

/// Platform clipboard access.
pub trait Clipboard {
    fn read(&mut self) -> Option<String>;
    fn write(&mut self, text: &str);
}

/// No-op clipboard for tests and headless use.
#[derive(Debug, Default)]
pub struct NoClipboard;

impl Clipboard for NoClipboard {
    fn read(&mut self) -> Option<String> {
        None
    }
    fn write(&mut self, _text: &str) {}
}

File: dotnet/Srui.Net/Core/Clipboard.cs
namespace Srui;

/// <summary>Platform clipboard access — injected by the host per the
/// sans-IO rule; every edit box gets it for free. A windowed app installs
/// the SDL system clipboard; a headless host installs its own with
/// <see cref="SruiApp.SetClipboard"/> (the default is a no-op).</summary>
public interface IClipboard
{
    string? Read();
    void Write(string text);
}

/// <summary>No-op clipboard for headless use without an injected one.</summary>
internal sealed class NoClipboard : IClipboard
{
    public string? Read() => null;
    public void Write(string text) { }
}
